aws-api-appsync/src/main/java/com/amplifyframework/api/aws/appsync/AppSyncGraphQLRequest.java

```diff
@@ -107,33 +105,29 @@ public void setVariable(String key, String type, String value) {
     }
 
     /**
-     * Used to add owner argument from authProvider if needed.
-     * @param authProvider CognitoUserPoolsAuthProvider for obtaining the username to set as the owner field.
-     * @throws ApiException if request requires owner argument and authProvider or authProvider.getUsername() is null.
+     * Returns whether owner argument is required for this request based on the model's {@link AuthRule}s.
+     * @return whether owner argument is required for this request based on the model's {@link AuthRule}s.
      */
-    public void setOwner(CognitoUserPoolsAuthProvider authProvider) throws ApiException {
+    public boolean isOwnerArgumentRequired() {
         for (AuthRule authRule : modelSchema.getAuthRules()) {
             if (isOwnerArgumentRequired(authRule)) {
-                setVariable(authRule.getOwnerFieldOrDefault(), "String!", getUsername(authProvider));
+                return true;
             }
         }
+        return false;
     }
 
-    private String getUsername(CognitoUserPoolsAuthProvider authProvider) throws ApiException {
-        if (authProvider == null) {
-            throw new ApiException(
-                    "Attempted to subscribe to a model with owner based authorization without a Cognito provider",
-                    "Did you add the AWSCognitoAuthPlugin to Amplify before configuring it?"
-            );
-        }
-        String username = authProvider.getUsername();
-        if (username == null) {
-            throw new ApiException(
-                    "Attempted to subscribe to a model with owner based authorization without a username",
-                    "Make sure that a user is logged in before subscribing to a model with owner based auth"
-            );
+    /**
+     * Used to add owner argument from authProvider if needed.
+     * @param owner username from CognitoUserPoolsAuthProvider to set as the owner field.
+     * @throws ApiException if request requires owner argument and authProvider or authProvider.getUsername() is null.
+     */
+    public void setOwner(String owner) {
```

Comment:
I tried to figure out a way to do that, but there's no good way to do it, without making breaking changes to public interfaces.  

Customers use the `ModelSubscription` factory to create a `GraphQLRequest` object, and then pass it to `Amplify.API.subscribe(...)`.    Then, in `AWSApiPlugin`, I call `setOwner` to add the username input.   The factory cannot do this unfortunately because it doesn't have a reference to the `CognitoUserPoolsProvider`.  Only the `AWSApiPlugin` has this reference.